core/integration/host_test.go

```diff
@@ -320,4 +320,61 @@ func (HostSuite) TestFile(ctx context.Context, t *testctx.T) {
 		require.NoError(t, err)
 		require.Equal(t, "hello world", content)
 	})
+	t.Run("cache behavior", func(ctx context.Context, t *testctx.T) {
```

Comment:
worth adding variations for `HostDirectoryOpts` too

Reply:
yep agreed

Reply:
done